### PR TITLE
test(sample): add opt-in deferred-init edge-case example (Android)

### PR DESCRIPTION
**Pull request by nickolas-dimitrakas**
## What

Adds a small, **disabled-by-default** example to the sample app that reproduces the Android *deferred-initialisation* edge case: starting mParticle from a native module at first-frame paint instead of `MainApplication.onCreate()`.

## Why

A partner deferring `MParticle.start()` (to shave init cost off startup) saw flaky Rokt overlay placements on Android but not iOS. Root cause: the Rokt Android SDK (≤ v5) caches the current `Activity` only on `onActivityResumed`, via an observer registered during `Rokt.init()` (which mParticle calls from `RoktKit.onKitCreate()`). When init runs **after** the host Activity has already resumed — exactly what deferred init does — that resume is missed, so overlay/bottom-sheet placements can't display until the next resume (the "press home and reopen" workaround). iOS resolves the presenter lazily at execute time, so it's unaffected.

Fixed upstream in the Rokt Android SDK: `ROKT/sdk-android-source` [#1062](https://github.com/ROKT/sdk-android-source/pull/1062) / [#1063](https://github.com/ROKT/sdk-android-source/pull/1063) (v5 backport #1082). This example is kept around as an easy way to re-trigger the scenario and confirm the fix / catch regressions.

## How it works

- `DeferredInitModule` (native) starts mParticle when JS calls it at first-frame paint (`requestAnimationFrame` in `index.js`).
- `ActivityTracker` mirrors Rokt's Activity-capture: an `EAGER` instance registered at process start vs a `DEFERRED` instance registered at init time. With the example on, `EAGER` captures `MainActivity` while `DEFERRED` stays `null` — the race, visible via `adb logcat -s DeferredInitRepro`.

## Default behaviour unchanged

`DeferredInitModule.DEFERRED_INIT_EXAMPLE` defaults to **`false`**, so the sample uses the standard eager init in `MainApplication.onCreate()`. Flip it to `true` to reproduce. See `sample/README.md` → "Deferred-init edge case (Android)".

## Validation

- `:app:compileDebugKotlin` passes (default flag off).
- Earlier, with the flag on, the race + the background/foreground recovery were reproduced on an emulator (logcat captured).

### CODE REVIEW

File: sample/index.js [hunk 1]
```diff
@@ -170,6 +171,23 @@ export default class MParticleSample extends Component {
...
+    // no-op. See DeferredInitModule.kt for the full explanation.
+    requestAnimationFrame(() => {
+      const {DeferredInit} = NativeModules;
+      if (DeferredInit) {
+        DeferredInit.startMParticle()
+          .then(r => console.log('Deferred MParticle.start ->', r))
+          .catch(e => console.warn('Deferred start failed', e));
+      } else {
+        console.warn('DeferredInit native module not available');
+      }
```
[the remark this answers is not in the document]
Comment (nickolas-dimitrakas) [reply sample/index.js]: Fixed in 52e1059 — the first-frame `DeferredInit` call is now gated behind `Platform.OS === 'android'`, so iOS no longer logs the spurious 'module not available' warning.

File: sample/index.js [hunk 2]
```diff
@@ -170,6 +171,23 @@ export default class MParticleSample extends Component {
...
+    });
```
[the remark this answers is not in the document]
Comment (nickolas-dimitrakas) [reply sample/index.js]: Fixed in 52e1059 — gated the block to Android (`Platform.OS === 'android'`); iOS stays quiet.

File: sample/android/app/src/main/java/com/mparticlesample/DeferredInitModule.kt [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    val identityRequest = IdentityApiRequest.withEmptyUser()
+    val options = MParticleOptions.builder(app)
+      .credentials("REPLACE_ME", "REPLACE_ME")
+      .logLevel(MParticle.LogLevel.VERBOSE)
+      .identify(identityRequest.build())
+      .build()
+
+    MParticle.start(options)
+    Log.i(TAG, "MParticle.start() invoked from native module. getInstance()=${MParticle.getInstance()}")
+
```
[the remark this answers is not in the document]
Comment (nickolas-dimitrakas) [reply sample/android/app/src/main/java/com/mparticlesample/DeferredInitModule.kt]: Fixed in 52e1059 — `startMParticle()` now short-circuits when `MParticle.getInstance() != null`, so repeat JS calls (Fast Refresh / remounts) won't double-register lifecycle callbacks or re-start the SDK.

File: sample/index.js [hunk 4]
```diff
@@ -170,6 +171,27 @@ export default class MParticleSample extends Component {
...
+        if (DeferredInit) {
+          DeferredInit.startMParticle()
+            .then(r => console.log('Deferred MParticle.start ->', r))
+            .catch(e => console.warn('Deferred start failed', e));
+        } else {
+          console.warn('DeferredInit native module not available');
+        }
+      });
+    }
+
```
[the remark this answers is not in the document]
Comment (nickolas-dimitrakas) [reply sample/index.js:194]: This is expected, by-design behaviour for this example rather than a bug: it's a second facet of the same deferred-init hazard the sample demonstrates — with init deferred to first-frame paint, mParticle JS calls made earlier (`Identity.login`, `getSession`) run before the SDK is started and no-op until then. Two mitigating points: the flag is **off by default** (mParticle starts eagerly in `onCreate()`, so these calls behave normally), and those calls are pre-existing sample code. Rather than plumb the native-only flag into JS and reorder the sample (which would obscure the race the example exists to show), I documented this as intentional in `sample/README.md` (5c643ac) with guidance to gate such calls behind init completion in a real deferred-init integration.

**Review by BrandonStalnaker — APPROVED** [review]
